language/puppet-functions.md

```diff
@@ -7,16 +7,20 @@ Grammar
 ---
 
     FunctionDefinition
-      : 'function' parameters = ParameterList? '{' statements += statements '}'
+      : 'function' name = NAME parameters = ParameterList? ReturnType? '{' statements += statements '}'
```

Comment:
My only concern here is consistency. People not familiar with this type of syntax will find formal description harder to read when the syntax varies depending on what's included in the text or not.